AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/HierarchicalVersionUtils.dfy

```diff
@@ -146,6 +146,13 @@ module {:options "/functionSyntax:4" } HierarchicalVersionUtils {
     map entry <- transformedContext :: entry.0 := entry.1
   }
 
+  predicate IsValidHV2EC?(branchKeyContext: Types.EncryptionContextString) {
+    && Structure.PrefixedEncryptionContext?(branchKeyContext - Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES)
+    && HasUniqueTransformedKeys?(branchKeyContext)
```

Comment:
Logically, I could have removed requires HasUniqueTransformedKeys?(branchKeyContext) but verification does not understand that when Structure.PrefixedEncryptionContext?(branchKeyContext - Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES) is true, HasUniqueTransformedKeys?(branchKeyContext) should be true and fails with "key expressions might be referring to the same value" on map entry <- transformedContext :: entry.0 := entry.1